isHierarchicalNamespaceEnabled check for Azure File System fails due to location

I have been trying to connect Trino to use delta tables from a Microsoft Fabric Lakehouse and keep getting an error message "Failed checking table location" (see below).

After investigating this error, I found that the `isHierarchicalNamespaceEnabled` method is the root cause of this failure.
If the `blobName` passed to the `getBlobClient` method is `/`, the function will error out with a 400 Bad Request error.
By changing the `blobName` to use the `location.directoryPath()`, the method succeeds and returns the correct response of `true` for the `isHierarchicalNamespaceEnabled` method.

Changing the directory from my testing has no impact with the results of the method as I tested this with`wasbs` and `abfss` urls for Azure Storage accounts with and without hierarchical namespace enabled and always received the correct response.

```
2025-03-28T08:19:49.344Z       DEBUG   dispatcher-query-5      io.trino.security.AccessControl Invocation of checkCanSetUser(principal=Optional[ubuntu], userName='ubuntu') succeeded in 26.10us
2025-03-28T08:19:49.344Z       DEBUG   dispatcher-query-5      io.trino.security.AccessControl Invocation of checkCanExecuteQuery(identity=Identity{user='ubuntu', principal=ubuntu}, queryId=20250328_081949_00002_9zbf8) succeeded in 10.00us
2025-03-28T08:19:49.347Z       DEBUG   dispatcher-query-2      io.trino.execution.QueryStateMachine    Query 20250328_081949_00002_9zbf8 is QUEUED
2025-03-28T08:19:49.348Z       DEBUG   dispatcher-query-5      io.trino.execution.QueryStateMachine    Query 20250328_081949_00002_9zbf8 is WAITING_FOR_RESOURCES
2025-03-28T08:19:49.349Z       DEBUG   dispatcher-query-5      io.trino.execution.QueryStateMachine    Query 20250328_081949_00002_9zbf8 is DISPATCHING
2025-03-28T08:19:49.349Z       DEBUG   dispatcher-query-1      io.trino.execution.QueryStateMachine    Query 20250328_081949_00002_9zbf8 is RUNNING
2025-03-28T08:19:49.362Z       DEBUG   Query-20250328_081949_00002_9zbf8-183   io.trino.security.AccessControl Invocation of checkCanExecuteProcedure(context=SecurityContext{identity=Identity{user='ubuntu', principal=ubuntu}, queryId=20250328_081949_00002_9zbf8}, procedureName=delta.system.register_table) succeeded in 768.17us
2025-03-28T08:19:49.366Z       DEBUG   Query-20250328_081949_00002_9zbf8-183   io.trino.security.AccessControl Invocation of checkCanCreateTable(context=SecurityContext{identity=Identity{user='ubuntu', principal=ubuntu}, queryId=20250328_081949_00002_9zbf8}, tableName=delta.default.fact_transaction, properties={}) succeeded in 1.24ms
2025-03-28T08:19:49.392Z       DEBUG   Query-20250328_081949_00002_9zbf8-183   io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastoreClient Invocation of getDatabase(name='default') succeeded in 23.44ms
2025-03-28T08:19:49.457Z       INFO    Query-20250328_081949_00002_9zbf8-183   com.azure.core.util.tracing.DefaultTracerProvider       Found Tracer implementation on the classpath: com.azure.core.tracing.opentelemetry.OpenTelemetryTracer
2025-03-28T08:19:51.793Z       ERROR   Query-20250328_081949_00002_9zbf8-183   com.azure.storage.blob.specialized.BlobClientBase       Status code 400, BadRequest
2025-03-28T08:19:51.797Z       DEBUG   dispatcher-query-1      io.trino.execution.QueryStateMachine    Query 20250328_081949_00002_9zbf8 is FAILED
2025-03-28T08:19:51.797Z       DEBUG   Query-20250328_081949_00002_9zbf8-183   io.trino.execution.QueryStateMachine    Query 20250328_081949_00002_9zbf8 failed
io.trino.spi.TrinoException: Failed checking table location abfss://xxx@onelake.dfs.fabric.microsoft.com/xxx.Lakehouse/Tables/xxx
       at io.trino.plugin.deltalake.procedure.RegisterTableProcedure.doRegisterTable(RegisterTableProcedure.java:164)
       at io.trino.plugin.deltalake.procedure.RegisterTableProcedure.registerTable(RegisterTableProcedure.java:122)
       at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
       at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:760)
       at io.trino.execution.CallTask.execute(CallTask.java:210)
       at io.trino.execution.CallTask.execute(CallTask.java:70)
       at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:152)
       at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:272)
       at io.trino.dispatcher.LocalDispatchQuery.startExecution(LocalDispatchQuery.java:150)
       at io.trino.dispatcher.LocalDispatchQuery.lambda$waitForMinimumWorkers$2(LocalDispatchQuery.java:134)
       at io.airlift.concurrent.MoreFutures.lambda$addSuccessCallback$12(MoreFutures.java:570)
       at io.airlift.concurrent.MoreFutures$3.onSuccess(MoreFutures.java:545)
       at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1139)
       at io.trino.$gen.Trino_474____20250328_081821_2.run(Unknown Source)
       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
       at java.base/java.lang.Thread.run(Thread.java:1447)
Caused by: java.io.IOException: Checking whether hierarchical namespace is enabled for the location abfss://xxx@onelake.dfs.fabric.microsoft.com/xxx.Lakehouse/Tables/xxx/_delta_log failed
       at io.trino.filesystem.azure.AzureFileSystem.isHierarchicalNamespaceEnabled(AzureFileSystem.java:601)
       at io.trino.filesystem.azure.AzureFileSystem.listFiles(AzureFileSystem.java:298)
       at io.trino.filesystem.switching.SwitchingFileSystem.listFiles(SwitchingFileSystem.java:116)
       at io.trino.filesystem.tracing.TracingFileSystem.lambda$listFiles$4(TracingFileSystem.java:119)
       at io.trino.filesystem.tracing.Tracing.withTracing(Tracing.java:51)
       at io.trino.filesystem.tracing.TracingFileSystem.listFiles(TracingFileSystem.java:119)
       at io.trino.plugin.deltalake.procedure.RegisterTableProcedure.doRegisterTable(RegisterTableProcedure.java:159)
       ... 16 more
Caused by: com.azure.storage.blob.models.BlobStorageException: Status code 400, BadRequest
       at com.azure.storage.blob.implementation.util.ModelHelper.mapToBlobStorageException(ModelHelper.java:660)
       at com.azure.storage.blob.implementation.BlobsImpl.getPropertiesNoCustomHeadersWithResponse(BlobsImpl.java:2571)
       at com.azure.storage.blob.specialized.BlobClientBase.lambda$existsWithResponse$3(BlobClientBase.java:675)
       at com.azure.storage.common.implementation.StorageImplUtils.sendRequest(StorageImplUtils.java:494)
       at com.azure.storage.blob.specialized.BlobClientBase.existsWithResponse(BlobClientBase.java:677)
       at com.azure.storage.blob.specialized.BlobClientBase.exists(BlobClientBase.java:653)
       at io.trino.filesystem.azure.AzureFileSystem.isHierarchicalNamespaceEnabled(AzureFileSystem.java:598)
       ... 22 more
```

--- a/lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java
+++ b/lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java
@@ -593,7 +593,7 @@ public class AzureFileSystem
     {
         try {
             BlockBlobClient blockBlobClient = createBlobContainerClient(location, Optional.empty())
-                    .getBlobClient("/")
+                    .getBlobClient(location.directoryPath())
                     .getBlockBlobClient();
             return blockBlobClient.exists();
         }

